moveit_experimental/moveit_jog_arm/src/jog_calcs.cpp

```diff
@@ -382,9 +385,19 @@ bool JogCalcs::jointJogCalcs(const control_msgs::JointJog& cmd, JogArmShared& /*
 
   prev_joint_velocity_ = delta_theta_ / parameters_.publish_period;
 
+  publishStatus();
+  // Cache the status so it can be retrieved asynchronously
+  updateCachedStatus(shared_variables, mutex);
+
   return convertDeltasToOutgoingCmd();
 }
 
+void JogCalcs::updateCachedStatus(JogArmShared& shared_variables, std::mutex& mutex)
```

Comment:
Why are you still passing the mutex around? I don't see it being used anywhere. Please remove it from `jointJogCalcs()` and `updateCachedStatus()`. In general, I would really prefer having a member struct/class that provides thread-save access to all shared member variables using getters and setters. Of course, atomic members are fine as well if applicable. Having mutexes everywhere and passing them around makes this code harder to read and maintain than it has to be.

Reply:
Sorry, I simply forgot to delete the mutex arguments. Good catch.

I also want to reduce mutex usage. Maybe in a follow-up PR?

Reply:
Follow-up PR is fine! Thanks for fixing